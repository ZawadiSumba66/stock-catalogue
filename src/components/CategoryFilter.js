import filterStock from '../actions/filter';
import store from '../store';

const CATEGORIES = ['ALL', 'Gainers', 'Losers'];

const CategoryFilter = () => {
  const handleFilterChange = (e) => {
    store.dispatch(filterStock(e.target.value));
  };
  return (
    <form className="form" data-testid="category-filter">
      Filter Stock:
      {' '}
      <select onChange={handleFilterChange} className="category-input">
        {
                CATEGORIES.map((value) => (
                  <option value={value} key={value}>
                    {value}
                  </option>
                ))
            }
      </select>
    </form>
  );
};

export default CategoryFilter;
