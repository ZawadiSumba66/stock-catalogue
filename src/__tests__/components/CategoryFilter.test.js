import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import CategoryFilter from '../../components/CategoryFilter';

configure({ adapter: new Adapter() });

it('contains filter Stock', () => {
  const component = shallow(<CategoryFilter />);
  expect(component.find('Filter Stock')).toBeTruthy();
});

it('matches the category filter snapshot', () => {
  const tree = renderer.create(<CategoryFilter />).toJSON();
  expect(tree).toMatchSnapshot();
});
