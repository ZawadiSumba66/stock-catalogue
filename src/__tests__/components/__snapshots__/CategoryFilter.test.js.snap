// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the category filter snapshot 1`] = `
<form
  className="form"
  data-testid="category-filter"
>
  Filter Stock:
   
  <select
    className="category-input"
    onChange={[Function]}
  >
    <option
      value="ALL"
    >
      ALL
    </option>
    <option
      value="Gainers"
    >
      Gainers
    </option>
    <option
      value="Losers"
    >
      Losers
    </option>
  </select>
</form>
`;
